get working model of color changing map up and running

Adelle-Pitsas opened the pull request:
This adds conditional color to the map points based on whether or not that sensor detects notable water levels.
![Screenshot 2024-04-01 at 12 08 26 PM](https://github.com/gulfofmaine/Neracoos-1-Buoy-App/assets/108096652/bc62f734-3afe-47b9-8b8d-c78b4591bda6)

![Screenshot 2024-04-01 at 12 09 09 PM](https://github.com/gulfofmaine/Neracoos-1-Buoy-App/assets/108096652/7a674b38-2cc6-4b32-b064-9320a95cc604)

Adelle-Pitsas commented on src/Features/ERDDAP/utils/waterLevelThresholds.tsx:
```diff
@@ -0,0 +1,51 @@
+  }
+  return null
+}
+
```
Theoretically, a way this can be calculated differently is by using an iterative recursive function. Iterating through the flood levels, and returning the flood level where the value is above the min_value but below the next thresholds min_value. However, this would assume the flood thresholds are in order within the array. This current way seems verbose, but also protects against out-of-order flood thresholds.

Adelle-Pitsas commented:
I'm not a fan of the colors I chose, so going to revisit those to make it look less chaotic

youngmorse commented:
This is looking great! I think we can fine tune the colors and marker shape/size once it feels like it's working how we want it too. I agree with Alex, we want to be consistent with NOAA's colors. But maybe we can take some artistic license and tone down the shade/intensity of electric green, fire alarm red, and blaze orange.

Adelle-Pitsas commented:
Great. Thanks for all that feedback! Ended up just copying the colors which we can change later.
Other things:
- Changed `alert` names to `threshold` so we don't go crazy
- Added a makeshift legend in the map
- Changed the plot band colors on the graph accordingly
<img width="1723" alt="Screenshot 2024-04-02 at 10 59 07 AM" src="https://github.com/gulfofmaine/Neracoos-1-Buoy-App/assets/108096652/54e42f32-a89c-4858-ac37-028073ebb0af">
